### PR TITLE
aip-1: Proposal Process Draft PR

### PR DESCRIPTION
**Pull request by activescott**
Submitting this for discussion (via comments on specific lines) in case someone has burning concerns or other feedback. It is still an incomplete draft, so do not feel compelled to discuss/comment yet. 

**Do not** merge this as a Draft until the following are checked off:
- [x] Scott offers complete draft 1
- [x] Scott requests merge of Draft (which I will when it is complete and others have provided some comments/feedback).

**I am requesting a merge of the draft**. Please review, edit, and offer feedback so that we can keep it moving. Thanks!

### CODE REVIEW

**Comment by activescott** [comment]
this is related to #1

File: proposals/aip-1.md
Comment (JonathonDunford) [inline proposals/aip-1.md]: This paragraph is a little confusing. I would change it to:

"An AIP is a document providing information to the Agorex community that describes a new process or feature within the organizational structure or the exchange itself. The AIP should provide a concise specification of the feature or process and a rationale for the feature. The AIP author is responsible for building consensus within the community and documenting dissenting opinions."

File: proposals/aip-1.md
Comment (JonathonDunford) [inline proposals/aip-1.md]: Rationale*

File: proposals/aip-1.md
Comment (JonathonDunford) [inline proposals/aip-1.md]: We should have a feature or exchange proposal outlined here as well.

Or are we keeping AIPs **only** for organizational matters?

File: proposals/aip-1.md
Comment (rowmagnon) [reply proposals/aip-1.md]: Agreed. Adds more clarity.

File: proposals/aip-1.md
Comment (JonathonDunford) [reply proposals/aip-1.md]: I would think this AIP process should be allowed for both, but with the exchange, speed of development is also important, so maybe a less formal process. (approved review of 1 board member for merges into master?)

File: proposals/aip-1.md
Comment (JonathonDunford) [inline proposals/aip-1.md]: Using different branches for this as long as it is well documented is one option.

I may be misunderstanding your question though.

When should members submit a review that "approve"s a PR? Only when we feel it should be Accepted?

File: proposals/aip-1.md
Comment (JonathonDunford) [inline proposals/aip-1.md]: What should the base readme include?
A summarized version of AIP1 and what this repo is?

File: proposals/aip-1.md
Comment (JonathonDunford) [inline proposals/aip-1.md]: Kind of an unnecessary pain.
We could just require 2 forms of approval: PR review approval + Verified Discord / Email / etc. approval

File: proposals/aip-1.md
Comment (activescott) [reply proposals/aip-1.md]: @JonathonDunford @rowmagnon Thank you for the feedback. I'm fine with this revision. However, I wonder if we want to include "feature" at all though (or at least maybe not now). I found the EIP wording to be fairly unclear for processes. It appears to me the doc was written for features and as an afterthought processes/meta was worked in. I have been thinking of optimizing the wording of AIP-1 to be "process" specific (and toyed with wording of "policy" too). I think feature decisions generally don't require the formality of an AIP and maybe we should focus on processes/policies instead for now. Then later we could add in features if we find that helpful. What are your thoughts?

File: proposals/aip-1.md
Comment (activescott) [reply proposals/aip-1.md]: Ah, yes... This is what I was referring to above. As I wrote [here](https://github.com/Agorex-io/Proposals/pull/3/files#diff-cb86b9455b45e60922306503f26e5c70R102) I think we should consider optimizing the document for process/organizational proposals **only**. I think proposals for new "features" on the exchange should be done via pull requests and issues. For minor features, just do it (or at least start it) and submit a PR. For more significant changes, start an issue to discuss it and get community feedback first. 

I think the challenge for a community, open source exchange at first is actually getting enough actual contributors to develop the features, and ideas and agreement on features is much easier. Once we have so many contributors willing to develop features and the more evident problem is just deciding which ones to do, we could formalize the process by updating this doc (with a new AIP, or a PR updating this one).

Ethereum is a bit different in that Ethereum is a protocol and enhancements and changes _require_ of clients (similar to python). I think for the most part the exchange won't have that as such a significant concern.

File: proposals/aip-1.md
Comment (activescott) [reply proposals/aip-1.md]: EIPs (and the python enhancement proposal that it is derived from) leave them so that people can see the past decision that was rejected for future reference. I think that makes sense.

File: proposals/aip-1.md
Comment (activescott) [reply proposals/aip-1.md]: This was more of a note to myself, and I think is now resolved with the "Acceptance Process" section. I will remove this and we can discuss the Acceptance Process section now.

File: proposals/aip-1.md
Comment (activescott) [reply proposals/aip-1.md]: Yes that is my intent. Just housekeeping.

File: proposals/aip-1.md
Comment (activescott) [reply proposals/aip-1.md]: I tend to agree, but this was suggestion from the community to consider so I thought I'd would include it here to allow the community to consider it further. We crypto people tend to lean towards the tinfoil hat side of things after all. 😄

File: proposals/aip-1.md
Comment (JonathonDunford) [reply proposals/aip-1.md]: Okay, so we will need labels for these different states so that we can filter them out.

File: proposals/aip-1.md
Comment (JonathonDunford) [reply proposals/aip-1.md]: Yeah, good inclusion.

I think we require 2FA on all GH accounts, an approved review, then one confirmation from outside source (Discord, email, etc)

That should cover our bases.

File: proposals/aip-1.md
Comment (JonathonDunford) [inline proposals/aip-1.md]: These all need 2 spaces at the end of the line or they wrap onto the same line.

File: proposals/aip-1.md
Comment (JonathonDunford) [inline proposals/aip-1.md]: Should be:

"The motivation should explain what motivated..."

File: proposals/aip-1.md
Comment (JonathonDunford) [inline proposals/aip-1.md]: If this is here, the "copyright/public domain" line above can be removed.

File: proposals/aip-1.md
Comment (JonathonDunford) [inline proposals/aip-1.md]: We need something down here talking about how this proposal process is largely derived from...

File: proposals/aip-1.md
Comment (JonathonDunford) [inline proposals/aip-1.md]: Should this be the duty of the editor or the submitter?

File: proposals/aip-1.md
Comment (JonathonDunford) [inline proposals/aip-1.md]: In the proposal editor section, it states that the editor will "accept the corresponding pull request". This section seems to state the obvious? Should they mark it as complete, notify a board member, and the board member can pull?

File: proposals/aip-1.md
Comment (JonathonDunford) [inline proposals/aip-1.md]: This markdown simply sets those links.

You'll also need to list them like so:

```
[the Issues section of this repository]
[the Agorex subreddit]
[the Agorex Discord chat rooms]
[pull request in this repository]
```

**Comment by JonathonDunford** [comment]
I also think that suggestions (like additions, edits, etc.) that get merged into the AIP (even this one) should reference those users in some way.

Should there be a separate section for that or should it be in the authors section at the top?

File: proposals/aip-1.md
Comment (activescott) [reply proposals/aip-1.md]: Yes, the status is in the preamble.

File: proposals/aip-1.md
Comment (activescott) [reply proposals/aip-1.md]: Good one. Fixed.

File: proposals/aip-1.md
Comment (activescott) [reply proposals/aip-1.md]: Fixed...

File: proposals/aip-1.md
Comment (activescott) [reply proposals/aip-1.md]: Fixed.

File: proposals/aip-1.md
Comment (activescott) [reply proposals/aip-1.md]: Doh! Fixed.

File: proposals/aip-1.md
Comment (activescott) [reply proposals/aip-1.md]: Editors reviews it. They _If the AIP isn't ready, an editor will send it back to the **author** for revision..._ Generally speaking the author will actually make the changes based on feedback from the editor.

File: proposals/aip-1.md
Comment (activescott) [reply proposals/aip-1.md]: I reworded the proposal editor section slightly to be more explicit and removed this Assumptions section. The Assumptions section was really more my notes and comments for you, which were later addressed in the core doc so they became repetitive. I don't think appropriate to leave in the doc permanently anyway.

File: proposals/aip-1.md
Comment (activescott) [reply proposals/aip-1.md]: @JonathonDunford Right, they are all referenced in the doc above. Do you think it will be best to duplicate each reference as an explicit link so that they are visible in the markdown? Or maybe put a comment to explain why they're only visible in the source markdown itself? Or maybe just remove the "References" heading?

**Comment by activescott** [comment]
> I also think that suggestions (like additions, edits, etc.) that get merged into the AIP (even this one) should reference those users in some way.
> Should there be a separate section for that or should it be in the authors section at the top?

I added an explicit Acknowledgements section.

**Comment by CryptoSmileee** [comment]
I have now read through all the changes in the pull request and it all looks good to me. Very good work by you :)

File: proposals/aip-1.md
Comment (JonathonDunford) [reply proposals/aip-1.md]: So then it shouldn't include "Edit the AIP" as a duty of the editor?
Just reviewing the AIP and making suggestions.

File: proposals/aip-1.md
Comment (JonathonDunford) [reply proposals/aip-1.md]: Maybe just remove the references heading.

Everything should be viewed from Markdown IMO

**Comment by JonathonDunford** [comment]
@activescott 

After that references fix, I think this is good to go and will give my stamp of approval then :+1:

File: proposals/aip-1.md
Comment (activescott) [reply proposals/aip-1.md]: Yep, thanks. Changing this to _Work with the author/champion to edit the AIP for language..._ I don't think there is a problem if the editor graciously does some cosmetic improvements such as spelling and other housekeeping items to the doc itself. I think it is also fine for the editor to ask the author/champion to make required changes. So my intention is to leave it a little open for interpretation here.

**Comment by activescott** [comment]
Thanks for the feedback everyone. Please review again and let me know what else you see.

**Review by JonathonDunford — APPROVED** [review]
This is ready to go as an official Draft.

**Comment by activescott** [comment]
@rowmagnon Will be great if you can review and approve for merging as a draft. Great if you can merge it if you approve!

**Pull request merged**
